### ISSUE TITLE
Feature Request: Add support for clientDomain in BuildChallengeTx

### ISSUE THREAD

**Issue by eduartua** [stellar/go#3780]
### What problem does your feature solve?
The function BuildChallengeTx does not support to add the clientDomain and the clientSigningKey into the transaction built.
See [here](https://github.com/stellar/go/blob/1dd5bb02c4da/txnbuild/transaction.go#L880).
### What would you like to see?
As you can see, the python SDK supports them:
Look at the source code, specifically the func [build_challenge_transaction](https://github.com/StellarCN/py-stellar-base/blob/master/stellar_sdk/sep/stellar_web_authentication.py#L68)

It uses 
```
:param client_domain: The domain of the client application requesting authentication
:param client_signing_key: The stellar account listed as the SIGNING_KEY on the client domain's TOML file
```
The after this [line](https://github.com/StellarCN/py-stellar-base/blob/master/stellar_sdk/sep/stellar_web_authentication.py#L114) we have
```
if client_domain:
        if not client_signing_key:
            raise ValueError(
                "client_signing_key is required if client_domain is provided."
            )
        transaction_builder.append_manage_data_op(
            data_name="client_domain",
            data_value=client_domain,
            source=client_signing_key,
        )
```
### What alternatives are there?
Is it viable to add support for them? I can send in a PR.

/txnbuild

**Comment by leighmcculloch** [comment]
👀 It appear that the Go SDK has fallen behind supporting all the capabilities that SEP-10 defines. It's missing the changes introduced in SEP-10 v3.2.0, i.e. stellar/stellar-protocol@01e0d98f514b609fa759c51df7738d6853736d1d.

**Comment by JakeUrban** [comment]
You're right, we never discussed it. Support has been added to the [Python SDK](https://github.com/StellarCN/py-stellar-base/commit/d09162139c6c0ddbc400ed992ab94d0a33fd6fa1#diff-f23b0d2171336e28edba2066e741ba4cde2f9022036f3a76c1455aa69faa69fc) though.

The changes required are:
- Adding `client_domain` and `client_signing_key` optional parameters to the build-challenge-transaction function
- Checking for a ManageData op with 'client_domain' as the key and checking for a signature from the operation's source account if present
- Allowing the source of the 'client_domain' ManageData operation to not equal the server's account

I can create issues on the SDKs to implement this. Is there anything else we need to do before moving forward?

**Comment by eduartua** [comment]
Thanks @JakeUrban and @leighmcculloch Let me know the new issues to see if I can help.

**Comment by leighmcculloch** [comment]
I think this issue can serve as the issue for this repo.

The list of changes required looks good to me.
